vowpalwabbit/marginal.cc

```diff
@@ -67,8 +144,22 @@ void predict_or_learn(data& sm, LEARNER::base_learner& base, example& ec)
           uint64_t second_index = j.index() & mask;
           uint64_t key = second_index + ec.ft_offset;
           marginal& m = sm.marginals[key];
+
+	  if(sm.compete) { //now update weights, before updating marginals
+	    expert_pair& e = sm.expert_state[key];
+	    float regret1 = alg_loss - (m.first/m.second - label)*(m.first/m.second - label);
```

Comment:
The comment about loss seems to apply here as well.  Why adopt squared loss only?